### DISCUSSION TITLE
Is there an option to remove shadow from scanned documents

### DISCUSSION THREAD

**Discussion by Sicos1977** [dlemstra/Magick.NET#1015] [Help] [answered]
Since I'm a newbie with Magick.NET I have this question. Is there an option to remove shadow from a scanned document.

See the image below (I cant show the complete image because there is personal information on it). When I use OTSU thresholding on this then to lower shadow part turns black so I was wondering if there is an option to remove the shadow before using OTSU?

![image](https://user-images.githubusercontent.com/6692947/131803286-c31ca679-313f-4e98-9314-ecf13a9f44a0.png)

**Comment by dlemstra** [comment]
You probably want to ask that question here instead: https://github.com/ImageMagick/ImageMagick/discussions. I can then help you at a later moment to translate those commands into Magick.NET method calls if you need some help with.

**Comment by Sicos1977** [comment]
Going to do that, thanks.

It seems a very difficult problem to fix, also tested this with Tesseract OCR own image optimalisation and that also turned it to a block of black.

I use ImageMagick.Net to cleanup declarations that people sent in that they have photographed with their smartphone. I understand there is no 100% solution for everything because they sent in to much junk :-)

**Comment by Sicos1977** [comment, marked as answer]
```c#
        private void RemoveShadow(MagickImage image)
        {
            // convert 131828504-61b59e6a-25f8-405c-9366-6158baeccabe.png -negate -lat 20x20+10% -negate result.png
            image.Negate();
            image.AdaptiveThreshold(20, 20, new Percentage(10));
            image.Negate();
        }
        #endregion
```
